packages/smart-contracts/scripts-create2/contract-setup/setupERC20SwapToConversion.ts

```diff
@@ -23,29 +23,26 @@ export const setupERC20SwapToConversion = async (
   );
   await Promise.all(
     hre.config.xdeploy.networks.map(async (network) => {
-      let provider;
-      if (network === 'celo') {
-        provider = utils.getCeloProvider();
-      } else {
-        provider = utils.getDefaultProvider(network);
-      }
-      const wallet = new hre.ethers.Wallet(hre.config.xdeploy.signer, provider);
-      const signer = wallet.connect(provider);
-      const ERC20SwapToConversionConnected = await ERC20SwapToConversionContract.connect(signer);
-      const adminNonce = await signer.getTransactionCount();
-      const gasPrice = await provider.getGasPrice();
+      try {
+        let provider;
+        if (network === 'celo') {
+          provider = utils.getCeloProvider();
+        } else {
+          provider = utils.getDefaultProvider(network);
```

Comment:
shouldn't `getDefaultProvider` handle this condition?

Reply:
If I remember correctly, when we introduce the celo specific case with Yoann we wanted to make it part of the `getDefaultProvider` but there were some blockers.